### ISSUE TITLE
`default`, `export`, and `namespace` rules should be disabled in the `typescript` config.

### ISSUE THREAD

**Issue by Jercik** [import-js/eslint-plugin-import#2346]
If I understand correctly, TypeScript already highlights the same issues as [`import/default`](https://github.com/import-js/eslint-plugin-import/blob/main/docs/rules/default.md), [`import/export`](https://github.com/import-js/eslint-plugin-import/blob/main/docs/rules/export.md) and [`import/namespace`](https://github.com/import-js/eslint-plugin-import/blob/main/docs/rules/namespace.md) rules. If that's true, then these rules should be disabled in the `typescript` config similarly to `import/named` here: https://github.com/import-js/eslint-plugin-import/blob/68cea3e6b6fe5fd61e5cf2e2c7c0be9e8dc597cb/config/typescript.js#L25-L30

Please correct me if I'm wrong in my assumption. Otherwise, I can create a PR with the changes.

**Comment by ljharb** [comment]
That’s incorrect. I believe TS doesn’t check CJS files, or files without types.

**Comment by liamfd-titan** [comment]
Would it be within the scope of this project to create an extendable config for projects that are TS only, with these rules disabled? Something like `plugin:import/typescript-only`, alongside the current `plugin:import/typescript`.

**Comment by ljharb** [comment]
No projects are TS only, because all published code is CJS or ESM.

**Comment by tylerlaprade** [comment]
I came here for this same reason. `import/export` is NOT redundant with TypeScript, but `import/default`, `import/named`, `import/namespace`, and `import/no-unresolved` all are. Additionally, `import/default` doesn't seem to trigger when I expect it to (but TS always does in those places). Instead of excluding all these rules, I replaced `plugin:import/recommended` and `plugin:import/typescript` with `plugin:import/warnings` in my config and manually added `import/export` to my rules, since that was the only useful one in my project.

**Comment by soryy708** [comment]
Relates to:

- https://github.com/import-js/eslint-plugin-import/issues/3135

While this one issue is older, the other one has more activity, so I'll mark this one duplicate, to consolidate discussion.
